pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -2013,3 +2015,43 @@ func LoadReleaseImageVerifier(config *rest.Config) (verify.Interface, error) {
 	}
 	return verify.NewFromManifests([]manifest.Manifest{m}, sigstore.NewCachedHTTPClientConstructor(sigstore.DefaultClient, nil).HTTPClient)
 }
+
+// IsErrorUpdateEvent returns true when the update event is from
+// error state in clusterdeployment.
+func IsErrorUpdateEvent(evt event.UpdateEvent) bool {
+	new, ok := evt.ObjectNew.(*hivev1.ClusterDeployment)
+	if !ok {
+		return false
+	}
+	if len(new.Status.Conditions) == 0 {
+		return false
+	}
+
+	old, ok := evt.ObjectOld.(*hivev1.ClusterDeployment)
+	if !ok {
+		return false
+	}
+
+	for _, cond := range new.Status.Conditions {
+		if controllerutils.IsConditionInDesiredState(cond) {
+			continue
+		}
+
+		oldcond := controllerutils.FindClusterDeploymentCondition(old.Status.Conditions, cond.Type)
+		if oldcond == nil {
+			return true // newly added condition in failed state
```

Comment:
failed state - or conditions were just initialized and set to unknown

Reply:
too specific for comment to be useful long term. unknown is undesired state for this context.